examples/llm/common/web_scraper_module.py

```diff
@@ -0,0 +1,144 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+
+import requests
+import requests_cache
+from bs4 import BeautifulSoup
+
+logging
+
+import logging
+from functools import partial
+
+import mrc
+import mrc.core.operators as ops
+import pandas as pd
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from pydantic import ValidationError
+
+import cudf
+
+from morpheus.messages import MessageMeta
+from morpheus.modules.schemas.examples.llm.web_scraper_schema import WebScraperSchema
+from morpheus.utils.module_utils import ModuleLoaderFactory
+from morpheus.utils.module_utils import register_module
+
+logger = logging.getLogger(__name__)
+
+WebScraperLoaderFactory = ModuleLoaderFactory("web_scraper", "morpheus_examples_llm", WebScraperSchema)
+
+
+def download_and_split(msg: MessageMeta, text_splitter, link_column, session) -> MessageMeta:
+    """
+Uses the HTTP GET method to download/scrape the links found in the message, splits the scraped data, and stores
+it in the output, excludes output for any links which produce an error.
+"""
+    if (link_column not in msg.get_column_names()):
+        return None
+
+    df = msg.copy_dataframe()
+
+    if isinstance(df, cudf.DataFrame):
+        df: pd.DataFrame = df.to_pandas()
+
+    # Convert the dataframe into a list of dictionaries
+    df_dicts = df.to_dict(orient="records")
+
+    final_rows: list[dict] = []
+
+    for row in df_dicts:
+        url = row[link_column]
+
+        try:
+            # Try to get the page content
+            response = session.get(url)
+
+            if (not response.ok):
+                logger.warning("Error downloading document from URL '%s'. " + "Returned code: %s. With reason: '%s'",
+                               url,
+                               response.status_code,
+                               response.reason)
+                continue
+
+            raw_html = response.text
+            soup = BeautifulSoup(raw_html, "html.parser")
+
+            text = soup.get_text(strip=True, separator=' ')
+            split_text = text_splitter.split_text(text)
+
+            for text in split_text:
+                row_cp = row.copy()
+                row_cp.update({"page_content": text})
+                final_rows.append(row_cp)
+
+            if isinstance(response, requests_cache.models.response.CachedResponse):
+                logger.debug("Processed cached page: '%s'", url)
+            else:
+                logger.debug("Processed page: '%s'", url)
+
+        except ValueError as exc:
+            logger.error("Error parsing document: %s", exc)
+            continue
+        except Exception as exc:
+            logger.error("Error downloading document from URL '%s'. Error: %s", url, exc)
+            continue
+
+    return MessageMeta(df=pd.DataFrame(final_rows))
```

Comment:
Since this requires a pandas DF, we should add a comment explaining why, and assert that C++ mode is disabled

Reply:
I don't think this actually requires Pandas. Even if it did, its going to immediately be cast to cudf. Fixing it.